README.md

```diff
@@ -1,86 +1,140 @@
+  - JS `TypedArray`s are mapped to C# `Memory<T>` and passed by reference using shared memory (no proxying is needed).
+  - Other types like enums, dates, and delegates are automatically marshaled as one would expect.
+  - Custom marshaling and marshaling hints [may be supported later](https://github.com/jasongin/napi-dotnet/pull/25).
+
+### Optional .NET native AOT compilation
+This library supports hosting the .NET Runtime in the same process as the JavaScript engine. Alternatively, it also supports building [native ahead-of-time (AOT) compiled C#](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/) libraries that are loadable as a JavaScript module _without depending on the .NET Runtime_.
+
+There are advantages and disadvantages to either approach:
+|                 | .NET Runtime | .NET Native AoT |
+|-----------------|--------------|-----------------|
+| API compatibility | Broad compatibility with .NET APIs | Limited compatibility with APIs designed to support AOT |
+| Ease of deployment | Requires a matching version of .NET to be installed on the target system | A .NET installation is not required (though some platform libs may be required on Linux/Mac)
+| Size of deployment | Compact - only IL assemblies need to be deployed | Larger due to building necessary runtime code - minimum 3 MB on Windows, ~13 MB on Linux |
+| Performance     | Slightly slower startup (JIT) | Slightly faster startup (no JIT) |
+| Runtime limitations | Full .NET functionality | Some .NET features like reflection and code-generation aren't supported |
+
+### High performance
+The project is designed to be as performant as possible when bridging between .NET and JavaScript. Techniques benefitting performance include:
+ - Automatic marshaling avoids any use of JSON serialization, and uses generated code to avoid reflection.
+ - Automatic marshalling uses shared memory or proxies when possible to minimize the amount of data transferred across the boundary.
+ - Simple calls between JS and C# require **_almost_** zero memory allocation. (Maybe it will be zero eventually.)
+ - Most JavaScript values are represented in C# as small structs (basically containing just a handle to the JS value), which helps avoid memory allocation.
+ - Marshaling code uses modern C# performance features like `Span<T>` and `stackalloc` to minimize heap allocations and copying.
+
+Thanks to these design choices, JS to .NET calls are [more than twice as fast](https://github.com/jasongin/napi-dotnet/pull/23) when compared to `edge-js` using [that project's benchmark](https://github.com/tjanczuk/edge/wiki/Performance).
+
+## Requirements
+ - .NET 6 or later
+    - .NET 7 or later is required for AoT support.
+ - Node.js v16 or later
+    - Other JS engines may be supported in the future.
+ - OS: Windows, Mac, or Linux
+    - It should work any platform where .NET 6 is supported.
```

Comment:
>.NET 6 [](http://example.com/codeflow?start=40&length=6)

Should we add CI tests for it?

Reply:
We should, because many people will be using .NET 6 for a while since it is LTS.

Reply:
Added to the backlog: https://github.com/users/jasongin/projects/1/views/1?pane=issue&itemId=21077422